### PR TITLE
Feature/prefab hidden folder

### LINKED ISSUE #120: Prefab saving issue with DefaultDiagnosticsWindow prefab
# XRTK - Mixed Reality Toolkit Bug Report

## Describe the bug

In a project consuming the 0.1.28 packages, the following error occurs in the Console when building.

```
Saving Prefab to immutable folder is not allowed: Packages/com.xrtk.sdk/Features/Diagnostics/Prefabs/DefaultDiagnosticsWindow.prefab
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
```

If you also open the Prefab from the editor, it locks up and spams the console with the following error:

```
ArgumentException: Saving Prefab to immutable folder is not allowed: 'Packages/com.xrtk.sdk/Features/Diagnostics/Prefabs/DefaultDiagnosticsWindow.prefab'
UnityEditor.PrefabUtility.ValidatePath (UnityEngine.GameObject instanceRoot, System.String path) (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.PrefabUtility.SaveAsPrefabAssetArgumentCheck (UnityEngine.GameObject instanceRoot, System.String path) (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.PrefabUtility.SaveAsPrefabAsset (UnityEngine.GameObject instanceRoot, System.String assetPath, System.Boolean& success) (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.Experimental.SceneManagement.PrefabStage.SavePrefab () (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.Experimental.SceneManagement.PrefabStage.SavePrefabWithVersionControlDialogAndRenameDialog () (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.Experimental.SceneManagement.PrefabStage.AutoSave () (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.Experimental.SceneManagement.PrefabStage.HandleAutoSave () (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.Experimental.SceneManagement.PrefabStage.Update () (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.SceneManagement.StageNavigationManager.ValidateAndUpdatePrefabStages (System.Boolean showDialogIfInvalid) (at <83a73882c51c4602b3d34743827d03e7>:0)
UnityEditor.SceneManagement.StageNavigationManager.Update () (at <83a73882c51c4602b3d34743827d03e7>:0)
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <437ba245d8404784b9fbab9b439ac908>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Delegate.DynamicInvokeImpl (System.Object[] args) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Delegate.DynamicInvoke (System.Object[] args) (at <437ba245d8404784b9fbab9b439ac908>:0)
UnityEditor.EditorApplication.Internal_CallUpdateFunctions () (at <83a73882c51c4602b3d34743827d03e7>:0)
```

## To Reproduce

Create a new project and import the XRTK.  Any attempt to build or "look" at the "DefaultDiagnosticsWindow" prefab causes the issue.

### PR DESCRIPTION
**Pull request by StephenHodgson**
https://github.com/XRTK/XRTK-Core/pull/563

- Fixes #120

### CODE REVIEW

File: XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/Diagnostics/Scripts/MixedRealityFrameDiagnosticsHandler.cs [hunk 1]
```diff
@@ -33,14 +35,33 @@ public class MixedRealityFrameDiagnosticsHandler : MonoBehaviour,
         [Tooltip("The text component used to display the GPU FPS information.")]
         private TextMeshProUGUI gpuFrameRateText = null;
 
+        [Prefab]
         [SerializeField]
-        [Tooltip("Image components used to visualize missed frames.")]
-        private Image[] missedFrameImages = null;
+        [Tooltip("Image component used to visualize missed frames.")]
```
Comment (FejZa) [inline XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/Diagnostics/Scripts/MixedRealityFrameDiagnosticsHandler.cs:40]: These changes seem unrelated to this PR

Comment (StephenHodgson) [reply XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/Diagnostics/Scripts/MixedRealityFrameDiagnosticsHandler.cs:40]: I mean I guess I could pull it out but originally I was trying to fix that prefab anyway and did lol.

Comment (FejZa) [reply XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/Diagnostics/Scripts/MixedRealityFrameDiagnosticsHandler.cs:40]: I am fine with not pulling it out I guess. It's gonna get in there anyway.
